x-pack/plugins/aiops/public/hooks/use_data.ts

```diff
@@ -91,12 +88,30 @@ export const useData = (
   ]);
 
   const _timeBuckets = useTimeBuckets();
-
   const timefilter = useTimefilter({
     timeRangeSelector: selectedDataView?.timeFieldName !== undefined,
     autoRefreshSelector: true,
   });
 
+  const fieldStatsRequest: DocumentStatsSearchStrategyParams | undefined = useMemo(() => {
+    const timefilterActiveBounds = timefilter.getActiveBounds();
+    if (timefilterActiveBounds !== undefined) {
+      _timeBuckets.setInterval('auto');
+      _timeBuckets.setBounds(timefilterActiveBounds);
+      _timeBuckets.setBarTarget(barTarget);
+      return {
+        earliest: timefilterActiveBounds.min?.valueOf(),
+        latest: timefilterActiveBounds.max?.valueOf(),
+        intervalMs: _timeBuckets.getInterval()?.asMilliseconds(),
+        index: selectedDataView.getIndexPattern(),
+        searchQuery,
+        timeFieldName: selectedDataView.timeFieldName,
+        runtimeFieldMap: selectedDataView.getRuntimeMappings(),
+      };
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [lastRefresh, searchQuery]);
```

Comment:
and afterwards `fieldStatsRequest` should be triggered based on explicit refresh or time range updates.